test/basicfuns.jl

```diff
@@ -149,7 +149,15 @@ end
 @testset "softmax" begin
     x = [1.0, 2.0, 3.0]
     r = exp.(x) ./ sum(exp.(x))
-    @test softmax([1.0, 2.0, 3.0]) ≈ r
+    @test softmax(x) ≈ r
     softmax!(x)
     @test x ≈ r
+    
+    x = [1, 2, 3]
+    r = exp.(x) ./ sum(exp.(x))
+    @test softmax(x) ≈ r
+    
+    x = [1//2, 2//3, 3//4]
+    r = exp.(x) ./ sum(exp.(x))
+    @test softmax(x) ≈ r
```

Comment:
Can you test the element type of `softmax(x)` in these examples and add a test with a `Vector{Float32}`?

Reply:
for rational and Int64 softmax will output Float64 (because of the float(eltype(x)) call) for Float32 is this acceptable behavior?

Reply:
Yep, this is the expected behaviour. The Float32 test is the most interesting one here I assume since currently it would just output a vector of Float64 in this case.